### PR TITLE
Removal of unnecessary lines and correction of Index numbers of Listings and Examples

### PR DESCRIPTION
**Pull request by sirnish**
*Issue #, if available:*
- There were many unnecessary lines, removed to make text more readable.
- Correction of Index numbers of Listings and Examples in few chapters

*Description of changes:*
Removal of unnecessary lines and correction of Index numbers of Listings and Examples

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ch02.md [hunk 1]
```diff
@@ -392,8 +392,8 @@ int main( void )
     return 0;
 }
 ```
-*Listing 2.1 The template for a new main() function*
-* * *
```
Comment (robhath) [inline ch02.md:396]: we have been placing figures, listings etc. between horizontal rules (lines)... was this meant to be deleted here?

File: ch03.md [hunk 2]
```diff
@@ -350,12 +350,12 @@ the call to `vPortDefineHeapRegions()`.
 
 <a name="list3.1" title="Listing 3.1 The vPortDefineHeapRegions() API function prototype"></a>
 
-* * *
```
Comment (robhath) [inline ch03.md:353]: was this deleted intentionally?

File: ch03.md [hunk 3]
```diff
@@ -350,12 +350,12 @@ the call to `vPortDefineHeapRegions()`.
...
+
 ```c
 void vPortDefineHeapRegions( const HeapRegion_t * const pxHeapRegions );
 ```
-*Listing 3.1 The vPortDefineHeapRegions() API function prototype*
-* * *
```
Comment (robhath) [inline ch03.md:358]: was this deleted intentionally?

File: ch04.md [hunk 4]
```diff
@@ -1855,7 +1812,7 @@ execute, and the resultant output is shown in Figure 4.15.
 
 ***
 ![](media/figure_4.14_example_4.8_execution_sequence.png)  
-*Figure 4.14 The sequence of task execution when running Example 4.8*
+***Figure 4.14** *The sequence of task execution when running Example 4.8*
```
Comment (robhath) [inline ch04.md:1815]: other instances use 3 *'s following the figure number

File: ch04.md [hunk 5]
```diff
@@ -1143,15 +1122,12 @@ function was called (as is the case with `vTaskDelay()`).
...
-***
-
 ```c
 void vTaskDelayUntil( TickType_t * pxPreviousWakeTime,
                       TickType_t xTimeIncrement );
 ```
 
-*Listing 4.14 vTaskDelayUntil() API function prototype*
-***
+***Listing 4.14** *vTaskDelayUntil() API function prototype*
```
Comment (xuelix) [inline ch04.md:1130]: Missing a * after 4.14

File: ch08.md [hunk 6]
```diff
@@ -19,13 +19,13 @@ Following are some examples:
   1. Task A executes and starts to write the string "Hello world" to the
      LCD.
 
-  1. Task A is pre-empted by Task B after outputting just the beginning
+  2. Task A is pre-empted by Task B after outputting just the beginning
```
Comment (xuelix) [inline ch08.md:22]: This change actually makes no difference.  Same with the below 3 changes

**Review by xuelix — APPROVED** [review]

**Review by robhath — COMMENTED** [review]
looks good

**Pull request merged**